optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/GizmoMemberAccessorEntityEnhancer.java

```diff
@@ -247,6 +237,37 @@ private static String getTypeDescriptor(java.lang.reflect.Type type) throws Clas
         }
     }
 
+    public static String generateGizmoInitializer(ClassOutput classOutput, Set<String> generatedClassNames) {
+        String generatedClassName = OptaPlannerGizmoInitializer.class.getName() + "$Implementation";
+        ClassCreator classCreator = ClassCreator
+                .builder()
+                .className(generatedClassName)
+                .interfaces(OptaPlannerGizmoInitializer.class)
+                .classOutput(classOutput)
+                .build();
+
+        classCreator.addAnnotation(ApplicationScoped.class);
+        MethodCreator methodCreator = classCreator.getMethodCreator(MethodDescriptor.ofMethod(OptaPlannerGizmoInitializer.class,
+                "setup", void.class));
+        ResultHandle memberAccessorMap = methodCreator.newInstance(MethodDescriptor.ofConstructor(HashMap.class));
+        for (String generatedMemberAccessor : generatedClassNames) {
+            ResultHandle generatedMemberAccessorResultHandle = methodCreator.load(generatedMemberAccessor);
+            ResultHandle memberAccessorInstance =
+                    methodCreator.newInstance(MethodDescriptor.ofConstructor(generatedMemberAccessor));
+            methodCreator.invokeInterfaceMethod(
+                    MethodDescriptor.ofMethod(Map.class, "put", Object.class, Object.class, Object.class),
+                    memberAccessorMap, generatedMemberAccessorResultHandle, memberAccessorInstance);
+        }
+        methodCreator.invokeStaticMethod(
+                MethodDescriptor.ofMethod(GizmoMemberAccessorFactory.class, "usePregeneratedMemberAccessorMap",
+                        void.class, Map.class),
+                memberAccessorMap);
+        methodCreator.returnValue(null);
+
+        classCreator.close();
```

Comment:
ClassCreator implements AutoCloseable, but this close() is't called when an exception occurs (which results in a memory leak).

Can we do ARM instead:

```
try (ClassCreator classCreator = ...) {

}
```

That will call close for us. Even if an exception occurs.